src/WorkflowCore/Services/WorkflowExecutor.cs

```diff
@@ -250,7 +247,11 @@ private async Task DetermineNextExecutionTime(WorkflowInstance workflow, Workflo
             workflow.Status = WorkflowStatus.Complete;
             workflow.CompleteTime = _datetimeProvider.UtcNow;
 
-            await _middlewareRunner.RunPostMiddleware(workflow, def);
+            using (var scope = _serviceProvider.CreateScope())
```

Comment:
Should we use `IScopeProvider` instead?

Reply:
Same here, since we do not have an `IStepExecutionContext`.